### DISCUSSION TITLE
Authorizing websocket connection with connectionParams

### DISCUSSION THREAD

**Discussion by devmentality0** [enisdenjo/graphql-ws#263] [Q&A] [answered]
Hello! 
I'm trying to connect my graphql server to remote schema following this example: https://www.graphql-tools.com/docs/remote-schemas#using-graphql-ws
My case is slightly more complex as the remote server requires authorization header with token. I tried to implement it this way:
```js
const wsExecutor: Executor = ({ document, variables, context }) => {
  const subscriptionClient = createClient({
    url: WS_ENDPOINT,
    webSocketImpl: ws,
    connectionParams: {
      headers: {
        authorization: context?.req.headers.authorization
      }
    }
  });

  return observableToAsyncIterable({
    ... /* the same as in the example */
  });
};
```
With that approach, I encountered a problem: connectionParams are not sent when the client connects to the server. 
I was expecting it to be passed somewhere here: https://github.com/enisdenjo/graphql-ws/blob/216075510ad7ecbe2fdd4ba6fe578a668ae38fe3/src/client.ts#L593

Is it intended behaviour? Is there any other way to pass authorization header to websocket connection? Thanks!

**Comment by enisdenjo** [comment, marked as answer]
You cannot provide custom headers to a WebSocket upgrade request.

The `connectionParams` is the [`payload` part of the `ConnectionInit` message](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#connectioninit) inside an established WebSocket connection and is available on the [server through the `Context`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.Context.md#connectionparams). Read the ["ws server and client auth usage with token expiration, validation and refresh" recipe](https://github.com/enisdenjo/graphql-ws#auth-token) for inspiration.

Furthermore, please also read: https://github.com/graphql/graphql-over-http/pull/140#issuecomment-703109328.
